packages/extension-api/src/extension-api.d.ts

```diff
@@ -183,12 +183,19 @@ declare module '@tmpwip/extension-api' {
     socketPath: string;
   }
 
+  export interface MachineInfo {
+    cpus: number;
+    memory: number;
+    diskSize: number;
+  }
+
   export interface ContainerProviderConnection {
     name: string;
     type: 'docker' | 'podman';
     endpoint: ContainerProviderConnectionEndpoint;
     lifecycle?: ProviderConnectionLifecycle;
     status(): ProviderConnectionStatus;
+    machineInfo?: MachineInfo;
```

Comment:
we can't add this information
 cpu, memory, diskSize are properties bound to the provider
https://github.com/containers/podman-desktop/blob/main/extensions/podman/package.json#L28-L49
it's per provider

Reply:
I don't get it. 
Those information are fetched by running this command
` const machineListOutput = await execPromise(getPodmanCli(), ['machine', 'list', '--format', 'json']);`
and that is related to the machines, not the provider. 
if i execute it on my cli i get
```
[
    {
        "Name": "test2",
        "Default": false,
        "Created": "2023-03-01T16:24:02.8246355+01:00",
        "Running": true,
        "Starting": false,
        "LastUp": "2023-03-01T16:26:45.4546807+01:00",
        "Stream": "35",
        "VMType": "wsl",
        "CPUs": 8,
        "Memory": "456134656",
        "DiskSize": "542113792",
        "Port": 56535,
        "RemoteUsername": "user",
        "IdentityPath": "C:\\Users\\Luca\\.ssh\\test2"
    },
    {
        "Name": "podman-machine-default",
        "Default": true,
        "Created": "2023-03-01T16:23:09.9128478+01:00",
        "Running": false,
        "Starting": false,
        "LastUp": "2023-03-01T16:23:09.9128478+01:00",
        "Stream": "35",
        "VMType": "wsl",
        "CPUs": 0,
        "Memory": "0",
        "DiskSize": "507510784",
        "Port": 56524,
        "RemoteUsername": "user",
        "IdentityPath": "C:\\Users\\Luca\\.ssh\\podman-machine-default"
    }
]
```
if i display those cpus/memory/disk values it seems reasonable.
Different things on linux which the `updateMachines` it's never called but i didn't manage it yet.

Reply:
a provider is "generic" so it can provide things or not
the list of properties that is provided is specified by the provider itself
you're listing me `podman machine info` command so it's specific to podman, not to all providers.

> if i display those cpus/memory/disk values it seems reasonable. They are specific infos about 2 podman machines.

it's podman specific, thus it can't be in the API
but it's exposed by properties of the podman extension

as previously linked https://github.com/containers/podman-desktop/blob/main/extensions/podman/package.json#L28-L49
here, podman extension says that it can provide cpu, memory, disk and some other stuff

but for Docker for example we don't provide this information 

so as of today, the resource component is "smart" and only displays stuff that extension is providing

You can display the "cpu, memory and disk" but those are not part of the API, but are properties that are exposed
so if CPU is exposed then you'll be able to display that information, etc.